src/lib.rs

```diff
@@ -1610,16 +1610,16 @@ impl Connection {
 
         // Create MAX_STREAM_DATA frames as needed.
         if pkt_type == packet::Type::Application && !is_closing {
-            for (id, stream) in self
-                .streams
-                .iter_mut()
-                .filter(|(_, s)| s.recv.more_credit())
-            {
+            for stream_id in self.streams.almost_full() {
+                let stream = self.streams.get_mut(stream_id).unwrap();
```

Comment:
I think we don't want the `unwrap()` in library code.

But also, looking at this a bit more deeply, it kinda sucks that `almost_full()` returns a `stream_id` iterator and not a `stream` iterator. I don't know if it's possible to do that inside stream.rs while also maintaining the optimized lookup that this PR introduces.

Reply:
Removed the unwrap().

As for the iterator thing, this is in part for consistency (so we use `StreamIter` everywhere instead of having different iterator in different cases) as well as because I think that returning stream directly might make the borrow checker sad.